fix(checker): indexed access (typeof Enum)["A"] reports spurious TS2339

## Summary

Indexing the type of an enum object with a string-literal key in type position, `(typeof E)["A"]`, fails in tsz with a spurious TS2339 ("Property 'A' does not exist on type 'E'"). tsc resolves it to the member type `E.A`.

The error message reveals the root cause: tsz is performing the indexed access against the enum *instance* type `E` rather than the enum *object* type `typeof E`.

## Minimal repro

`z3b.ts`:
```ts
enum E { A, B }
type T = (typeof E)["A"];
```

tsc (6.0.2, --strict): no errors (`T` = `E.A`).

tsz:
```
z3b.ts(2,21): error TS2339: Property 'A' does not exist on type 'E'.
```

## Expected

`(typeof E)["A"]` resolves to `E.A`, no diagnostic.

## Boundary / controls (all PASS in tsz — isolating the bug)

- `type V = (typeof E)[K]` where `K = keyof typeof E` works fine.
- Value position `const te = E; const a = te["A"];` works (`a: E.A`).
- Only the *string-literal* indexed access type `(typeof E)["A"]` fails.

So the failure is specific to: indexed-access **type** with a literal string key, where the object operand is `typeof Enum`.

## Root-cause hypothesis + layer

When evaluating `IndexedAccess(typeof E, "A")`, tsz appears to strip/normalize `typeof E` down to the enum instance type `E` before the property lookup, so the string-literal key `"A"` (a member of the enum object, not the instance) is not found. The `keyof typeof E` path works because it threads the object type through differently.

Layer: solver indexed-access evaluation over `typeof Enum` object types (the operand should retain the enum object/`typeof` type, not be reduced to the enum instance type).

## Test matrix

1. `(typeof E)["A"]` numeric enum (above).
2. `(typeof E)["A"]` string enum `enum E { A="a" }`.
3. Renamed enum/members.
4. `(typeof E)[K]` with `K = keyof typeof E` (negative/control — must keep working).
5. Value-position `te["A"]` (control — must keep working).

AgentName: enum-hunter

--- a/crates/tsz-checker/src/types/type_node_advanced/enum_indexed_access.rs
+++ b/crates/tsz-checker/src/types/type_node_advanced/enum_indexed_access.rs
@@ -2,8 +2,17 @@ use super::super::type_node::TypeNodeChecker;
 use tsz_solver::TypeId;
 
 impl<'a, 'ctx> TypeNodeChecker<'a, 'ctx> {
-    pub(super) fn enum_namespace_property_object(&self, type_id: TypeId) -> Option<TypeId> {
-        let sym_id = self.ctx.resolve_type_to_symbol_id(type_id)?;
+    pub(super) fn enum_namespace_property_object(
+        &self,
+        surface_type: TypeId,
+        resolved_type: TypeId,
+        surface_is_type_query_node: bool,
+    ) -> Option<TypeId> {
+        if !surface_is_type_query_node && !self.type_surface_is_type_query_alias(surface_type) {
+            return None;
+        }
+
+        let sym_id = self.ctx.resolve_type_to_symbol_id(resolved_type)?;
         let symbol = self.ctx.binder.symbols.get(sym_id)?;
         if !symbol.has_any_flags(tsz_binder::symbol_flags::ENUM)
             || symbol.has_any_flags(tsz_binder::symbol_flags::ENUM_MEMBER)
@@ -12,6 +21,40 @@ impl<'a, 'ctx> TypeNodeChecker<'a, 'ctx> {
         }
 
         self.ctx.enum_namespace_types.get(&sym_id).copied()
+    }
+
+    fn type_surface_is_type_query_alias(&self, type_id: TypeId) -> bool {
+        if crate::query_boundaries::common::is_type_query_type(self.ctx.types, type_id) {
+            return true;
+        }
+
+        let Some(def_id) = crate::query_boundaries::common::lazy_def_id(self.ctx.types, type_id)
+        else {
+            return false;
+        };
+        let Some(def) = self.ctx.definition_store.get(def_id) else {
+            return false;
+        };
+        if def.kind != tsz_solver::def::DefKind::TypeAlias {
+            return false;
+        }
+
+        def.symbol_id
+            .map(tsz_binder::SymbolId)
+            .and_then(|sym_id| self.ctx.binder.get_symbol(sym_id))
+            .is_some_and(|symbol| {
+                symbol.declarations.iter().any(|&decl_idx| {
+                    let Some(node) = self.ctx.arena.get(decl_idx) else {
+                        return false;
+                    };
+                    let Some(alias) = self.ctx.arena.get_type_alias(node) else {
+                        return false;
+                    };
+                    self.ctx.arena.get(alias.type_node).is_some_and(|body| {
+                        body.kind == tsz_parser::parser::syntax_kind_ext::TYPE_QUERY
+                    })
+                })
+            })
     }
 
     pub(super) fn full_enum_member_union_parent_type(&self, type_id: TypeId) -> Option<TypeId> {

--- a/crates/tsz-checker/src/types/type_node_advanced/enum_indexed_access.rs
+++ b/crates/tsz-checker/src/types/type_node_advanced/enum_indexed_access.rs
@@ -2,6 +2,18 @@ use super::super::type_node::TypeNodeChecker;
 use tsz_solver::TypeId;
 
 impl<'a, 'ctx> TypeNodeChecker<'a, 'ctx> {
+    pub(super) fn enum_namespace_property_object(&self, type_id: TypeId) -> Option<TypeId> {
+        let sym_id = self.ctx.resolve_type_to_symbol_id(type_id)?;
+        let symbol = self.ctx.binder.symbols.get(sym_id)?;
+        if !symbol.has_any_flags(tsz_binder::symbol_flags::ENUM)
+            || symbol.has_any_flags(tsz_binder::symbol_flags::ENUM_MEMBER)
+        {
+            return None;
+        }
+
+        self.ctx.enum_namespace_types.get(&sym_id).copied()
+    }
+
     pub(super) fn full_enum_member_union_parent_type(&self, type_id: TypeId) -> Option<TypeId> {
         let list_id = crate::query_boundaries::common::union_list_id(self.ctx.types, type_id)?;
         let members = self.ctx.types.type_list(list_id);

--- a/crates/tsz-checker/src/types/type_node_advanced/indexed_access_type.rs
+++ b/crates/tsz-checker/src/types/type_node_advanced/indexed_access_type.rs
@@ -404,7 +404,9 @@ impl<'a, 'ctx> TypeNodeChecker<'a, 'ctx> {
                     // Check property existence against the materialized object so
                     // unevaluated mapped-alias intersection members contribute their
                     // keys (see `evaluate_type_for_property_check`).
-                    let property_object = self.evaluate_type_for_property_check(resolved_object);
+                    let property_object = self
+                        .enum_namespace_property_object(resolved_object)
+                        .unwrap_or_else(|| self.evaluate_type_for_property_check(resolved_object));
                     let prop_result =
                         crate::query_boundaries::property_access::resolve_property_access(
                             self.ctx.types,

--- a/crates/tsz-checker/src/types/type_node_advanced/indexed_access_type.rs
+++ b/crates/tsz-checker/src/types/type_node_advanced/indexed_access_type.rs
@@ -405,7 +405,11 @@ impl<'a, 'ctx> TypeNodeChecker<'a, 'ctx> {
                     // unevaluated mapped-alias intersection members contribute their
                     // keys (see `evaluate_type_for_property_check`).
                     let property_object = self
-                        .enum_namespace_property_object(resolved_object)
+                        .enum_namespace_property_object(
+                            object_type,
+                            resolved_object,
+                            object_is_type_query_node,
+                        )
                         .unwrap_or_else(|| self.evaluate_type_for_property_check(resolved_object));
                     let prop_result =
                         crate::query_boundaries::property_access::resolve_property_access(

--- a/crates/tsz-checker/tests/enum_indexed_access_tests.rs
+++ b/crates/tsz-checker/tests/enum_indexed_access_tests.rs
@@ -34,6 +34,14 @@ fn expect_no_type_or_property_error(source: &str) {
     );
 }
 
+fn expect_property_error(source: &str) {
+    let errors = diagnostics_for(source)
+        .into_iter()
+        .filter(|&c| c == 2339)
+        .count();
+    assert!(errors > 0, "Expected at least one TS2339 error:\n{source}");
+}
+
 fn expect_error(source: &str) {
     let errors = diagnostics_for(source)
         .into_iter()
@@ -98,6 +106,28 @@ declare const write: WriteType;
 const check: Mode.Write = write;
 "#;
     expect_no_type_or_property_error(source);
+}
+
+#[test]
+fn test_numeric_enum_typeof_alias_indexed_access_literal_member() {
+    let source = r#"
+enum Direction { Up, Down, Left, Right }
+type DirectionObject = typeof Direction;
+type UpType = DirectionObject["Up"];
+declare const up: UpType;
+const check: Direction.Up = up;
+"#;
+    expect_no_type_or_property_error(source);
+}
+
+#[test]
+fn test_numeric_enum_type_alias_indexed_access_rejects_namespace_member() {
+    let source = r#"
+enum Direction { Up, Down, Left, Right }
+type DirectionValue = Direction;
+type UpType = DirectionValue["Up"];
+"#;
+    expect_property_error(source);
 }
 
 // ---------------------------------------------------------------------------

--- a/crates/tsz-checker/tests/enum_indexed_access_tests.rs
+++ b/crates/tsz-checker/tests/enum_indexed_access_tests.rs
@@ -9,16 +9,33 @@
 
 use tsz_checker::test_utils::check_source_codes;
 
+fn diagnostics_for(source: &str) -> Vec<u32> {
+    check_source_codes(source)
+}
+
 fn expect_no_error(source: &str) {
-    let errors = check_source_codes(source)
+    let errors = diagnostics_for(source)
         .into_iter()
         .filter(|&c| c == 2322)
         .count();
     assert_eq!(errors, 0, "Expected no TS2322 errors:\n{source}");
 }
 
+fn expect_no_type_or_property_error(source: &str) {
+    let diagnostics = diagnostics_for(source);
+    let unexpected = diagnostics
+        .iter()
+        .copied()
+        .filter(|&c| matches!(c, 2322 | 2339))
+        .collect::<Vec<_>>();
+    assert!(
+        unexpected.is_empty(),
+        "Expected no TS2322/TS2339 errors, got {unexpected:?} from {diagnostics:?}:\n{source}"
+    );
+}
+
 fn expect_error(source: &str) {
-    let errors = check_source_codes(source)
+    let errors = diagnostics_for(source)
         .into_iter()
         .filter(|&c| c == 2322)
         .count();
@@ -59,6 +76,28 @@ const g: number = obj["Green"];
 const b: number = obj["Blue"];
 "#;
     expect_no_error(source);
+}
+
+#[test]
+fn test_numeric_enum_typeof_indexed_access_literal_member() {
+    let source = r#"
+enum Direction { Up, Down, Left, Right }
+type UpType = (typeof Direction)["Up"];
+declare const up: UpType;
+const check: Direction.Up = up;
+"#;
+    expect_no_type_or_property_error(source);
+}
+
+#[test]
+fn test_numeric_enum_typeof_indexed_access_literal_renamed_member() {
+    let source = r#"
+enum Mode { Read = 1, Write = 2 }
+type WriteType = (typeof Mode)["Write"];
+declare const write: WriteType;
+const check: Mode.Write = write;
+"#;
+    expect_no_type_or_property_error(source);
 }
 
 // ---------------------------------------------------------------------------
@@ -114,6 +153,17 @@ declare const obj: typeof Status;
 const v: string = obj.Active;
 "#;
     expect_no_error(source);
+}
+
+#[test]
+fn test_string_enum_typeof_indexed_access_literal_member() {
+    let source = r#"
+enum Status { Active = "active", Inactive = "inactive" }
+type ActiveType = (typeof Status)["Active"];
+declare const active: ActiveType;
+const check: Status.Active = active;
+"#;
+    expect_no_type_or_property_error(source);
 }
 
 #[test]

